### DISCUSSION TITLE
Flycheck configuration with tide

### DISCUSSION THREAD

**Discussion by Aaronzinhoo** [ananthakumaran/tide#416] [Q&A] [answered]
Just a general question of how to configure tide and flycheck. I use flycheck-tide now and want to use typescript-eslint. Should I use flycheck-tide and assign eslint as the next checker or vice versa? I noticed I was doing it wrong once I realized typescript-tide does so much more than tslint (before I found out the issue with it). If this is a question for flycheck sorry about wasting your time.

**Comment by ananthakumaran** [comment, marked as answer]
yes, the checker name is `typescript-tide` it follows `{mode}-tide` pattern. For typescript/tsx, it's usually better to run the tide checker first (as it reports type errors) followed by eslint/tslint. For javascript, tide checker usually is not much helpful (unless checkJs is set to true), so typically you can just run eslint only.

tide by default only enables the checker for typescript/tsx. For js, we append to the end, so if you have eslint, it will be given preference.

```
(flycheck-add-next-checker 'typescript-tide '(warning . typescript-tslint) 'append)
```

https://www.flycheck.org/en/latest/user/syntax-checkers.html#configuring-checker-chains  has more details

**Reply by Aaronzinhoo** [reply]
Thanks for the link and help on such a basic question. Really appreciate your help overall and improving my understanding.

Since `typescript-tide` is loaded with tide, 

```
(flycheck-add-next-checker 'typescript-tide '(warning . typescript-tslint) 'append)
```
should be loaded within tide's config just to avoid any possible errors.
